src/PermissionRegistrar.php

```diff
@@ -312,7 +319,8 @@ private function getSerializedPermissionsForCache()
     {
         $this->except = config('permission.cache.column_names_except', ['created_at','updated_at', 'deleted_at']);
 
-        $permissions = $this->getPermissionClass()->select()->with('roles')->get()
+        $permissions = $this->getPermissionClass()->select()->with('roles')
+            ->when($this->activeGuard, fn($query) => $query->where('guard_name', $this->activeGuard))->get()
```

Comment:
FYI, this package is still compatible with PHP 7.3, but the `fn()` syntax is not. That's why the tests are failing.